### ISSUE TITLE
Multiple log files fail config parser: patched, but not incorporated?

### ISSUE THREAD

**Issue by dever23b** [fail2ban/fail2ban#2341]
This is in reference to #976, #727, #1721 and similar tickets.

Thanks for your work on fixing the problem!  I was very happy to see this issues has been resolved (#1622).  Unfortunately, I can't seem to find a release that contains the patch.  I realize that it's been fixed, but I'm a little confused about how to "get" the fix.

Forgive my ignorance: I don't have much experience with GitHub and I haven't been able to figure out how to trace this commit to a release.  I noticed that this commit closes several open tickets regarding multiple log paths and newlines.  I saw the comment that it's been fixed for 0.10 and it will not be fixed in 0.9.x; however, when I upgraded to 0.10.2-2 for Debian, the patches contained here are not in the package.  When I attempt to browse the 0.10 (or 0.11) branch on GitHub and compare jailreader.py, it doesn't seem like these changes got committed.  

I don't mean to accuse of anything: I'm admitting my ignorance using GitHub.  But, did these changes ever get incorporated?  If so, what version do I need to install?  I can't seem to find a release that contains this patch.

### Environment:
- Fail2Ban version (including any possible distribution suffixes): Fail2Ban v0.10.2
- OS, including release name/version: Raspbian GNU/Linux 9 (stretch)
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
I'm trying to configure fail2ban to monitor multiple logpaths for my web server setup.  The issue has been that fail2ban hasn't been able to handle multiple logpaths when they get interpolated into an action variable, and a syntax error develops (see #976, #727, #1721).  To overcome this, I've had to dump all of my virtual host logs in the same folder, using the syntax <domain>_(access|error).log.  This results in very long filenames, which makes accessing the logs from the command line tedious.  I'd like to organize my logs into subfolders named after each domain.  When I found #1622, I was led to believe that this issue has already been dealt with and patched and all I needed to do was update fail2ban.  Unfortunately, after updating, I discovered that the patch doesn't seem to have made it to the release available for my distribution.  I considered cloning the repository directly; however, I haven't been able to find a release that contains the patch.  

#### Steps to reproduce
See #976.   Basically, define a logpath with two paths separated by a line break, and then attempt to use the default action_mwl action.  Also persists using the recommendation to enclose %(logpath)s in quotes, in the action_mwl definition.

#### Expected behavior
Fail2Ban expands the wildcards in both log paths and monitors all relevant logs.  Relevant log data from the correct, corresponding log file, is also included in the notification that gets sent via the action_mwl action via %(mta)s-whois-lines.

#### Observed behavior
```
Invalid action definition 'sendmail-whois-lines[name=nginx-http-auth, sender="[redacted]", dest="[redacted]", logpath="/var/log/nginx/*access.log'
Errors in jail 'nginx-http-auth'. Skipping...
```
( Note the error terminating with a single quote after access.log, rather than a double quote where the path begins: it's breaking at the newline character in my variable for the log paths )

#### Any additional information
I apologize if this is the incorrect way to resurrect an issue, but I wasn't sure if anyone would be notified if I commented on the other related issues that have all been closed.

### Configuration, dump and another helpful excerpts
My action variables:
```
action_mwl = %(banaction)s[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
             %(mta)s-whois-lines[name=%(__name__)s, sender="%(sender)s", dest="%(destemail)s", logpath="%(logpath)s", chain="%(chain)s"]

action = %(action_mwl)s
```

My log path variables:
```
nginx_error_log     = /var/log/nginx/*error.log
                      /var/log/nginx/*/error.log
nginx_access_log    = /var/log/nginx/*access.log
                      /var/log/nginx/*/access.log
```

( Note: I tried using both tabs and spaces.  I also attempted to replace the newline with a space, but that caused a different failure )
#### Any customizations done to /etc/fail2ban/ configuration
See above

#### Relevant parts of /var/log/fail2ban.log file:
The log file does not contain any relevant information: the output noted above comes from STDOUT/STDERR of the `fail2ban-client reload` command.

#### Relevant lines from monitored log files in question:
N/A

_Edit: include OS information, formatting_

**Comment by sebres** [comment]
> when I upgraded to 0.10.2-2 for Debian, the patches contained here are not in the package

Hmm... Indeed. I checked stretch-backports, and fail2ban (0.10.2-2~bpo9+1) seems really miss that.
No idea why. The branch (as well as merge-commit 389ad103442a779a40516728e4cd6ae04ee42dde) is old, and corresponding git-log (and github too) included already since 0.10.0 release.
@yarikoptic, @fail2ban/maintainers Any idea how it could happen?

As for #976 and action_mwl...

Possibly this is caused because of the missing merge, but...
But I don't like actions like this - IMHO they are all obsolete, the reason is - parsing of log-files (via grep) to find matched lines containing failures, that are already available in the ticket self as well as within the action (with tags `<matches>` and `<ipmatches>`) is weird.
There are already several mail-actions that are using this tags, so it is better to use this.

**Comment by sebres** [comment]
As for "missing" merge in 0.10.2-2 - false alarm.
Compared `0.10.2-2~bpo9+1` with release `0.10.2` - both are totally equivalent. 
Just `jailreader.py` has been several times changed between 0.10.0 and 0.10.2 (and now too).

So to install current branch clone/download `0.10` (or `debian` branch, which is admittedly not always actual).

**Comment by sebres** [comment]
As for `action_mwl`, see - https://github.com/fail2ban/fail2ban/issues/976#issuecomment-462290167

Thus closed as fixed.

**Comment by dever23b** [comment]
Thanks so much for the quick response!  Just to confirm, do I understand correctly that this is just a commit on the existing version, so I need to uninstall the debian package and clone it instead?  (Trying to confirm the best way to receive the patch.)

Thanks again!

**Comment by sebres** [comment]
Yes, just I had it initially merged in 0.10/0.11 only... Now it is in debian branch too (as already said this branch is rather rarely up-to-date).
And of course only if you basically familiar with manual installation.
Otherwise wait until it gets released or still better try other mail-actions using tags `<matches>` and `<ipmatches>` (as mentioned above).

**Comment by dever23b** [comment]
Thanks much.  I'll look into the other actions and then, if I don't have any luck, clone the repo and attempt a manual installation.  Appreciate your help!
